js/src/launchAsync.ts

```diff
@@ -63,6 +74,273 @@ let launchResponseResolve: LaunchResponseResolve;
+                // No-op
+            }
+
+            if (response && response.success) {
+                const playMessageValue: any = {
+                    command: 'PlayState',
+                    parameters: 'Play'
+                };
+
+                const pauseMessageValue: any = {
+                    command: 'PlayState',
+                    parameters: 'Pause'
+                };
+
+                launchResponse = {
+                    container: iframeContainer,
+                    sessionId: response.sessionId,
+                    charactersProcessed: response.meteredContentSize,
+                    postLaunchOperations: {
+                        pause: () => {
+                            iframe.contentWindow!.postMessage(JSON.stringify({ messageType: 'InstrumentationCommand', messageValue: pauseMessageValue }), '*');
+                        },
+                        play: () => {
+                            iframe.contentWindow!.postMessage(JSON.stringify({ messageType: 'InstrumentationCommand', messageValue: playMessageValue }), '*');
+                        }
+                    }
+                };
+            } else if (response && !response.success) {
+                error = {
+                    code: response.errorCode,
+                    message: errorMessageMap[response.errorCode],
+                    sessionId: response.sessionId
+                };
+            } else {
+                error = {
+                    code: ErrorCode.ServerError,
+                    message: errorMessageMap[ErrorCode.ServerError]
+                };
+            }
+
+            isLoading = false;
+
+            if (funcType === FuncType.launchAsync) {
+                const {
+                    reject,
+                    resolve
+                } = extras;
+                if (launchResponse) {
+                    resetTimeout();
+                    resolve(launchResponse);
+                } else if (error) {
+                    exit(options.onExit);
+                    reject(error);
+                }
+            } else if (funcType === FuncType.launchWithoutContentAsync) {
+                if (launchResponse) {
+                    resetTimeout();
+                    if (launchResponseResolve) {
+                        launchResponseResolve(launchResponse);
+                        return;
+                    }
+                } else if (error) {
+                    exit(options.onExit);
+                    if (launchResponseReject) {
+                        launchResponseReject(error);
+                    }
+                }
+            }
+        } else if (e.data.startsWith(PostMessagePreferences)) {
+            if (options.onPreferencesChanged && typeof options.onPreferencesChanged === 'function') {
+                try {
+                    options.onPreferencesChanged(e.data.substring(PostMessagePreferences.length));
+                    return; // TODO: Yes?
```

Comment:
Please clarify.

Reply:
i wasnt sure if the change in the function setup would require a return here, but there aren't any subsequent calls that could happen as a side effect so i've removed the return